File: lab3/reverse.js
/***************************************************
 * Lab 3 -- Part 2: reverse.js 
 *  	
 * The goal of this task is to read in a sequence of integers read 
 * via STDIN and then print them to STDOUT in reverse order (one per line).  	
 *		  
 *************************************************		  
 */  	


var readline = require('readline');


/*************************************************  	
 * Honor pledge (please write your name.)		  
 *	
 * I **Scott Forrest** have completed this code myself, without
 * unauthorized assistance, and have followed the academic honor code.		  
 *
 * Edit the code below to complete your solution. 
 * 
 *************************************************** 
 */		  


var rl = readline.createInterface({
    input: process.stdin,  	
    output: process.stdout,	
    terminal: false  	
});  	

var nums  = []; 

rl.on('line',		  
      (chunk) =>
      {
          nums.push(chunk.trim());
      });  	


rl.on('close',
      () =>		  
      {
          nums.reverse((a, b)=>b-a);
          nums.map((n) => ""+n);
          console.log(nums.join("\n"))
          // TODO: print out nums, one per line
      });
		  
  	
		  

File: lab0/reverse.cpp
/***************************************************		  
 * Lab 0 -- Part 2: reverse.cpp 
 *  	
 * The goal of this task is to use a linked list to read in a sequence of integers
 * via STDIN and then print them to STDOUT in reverse order (one per line).	
 *		  
 *************************************************  	
 */ 

	
#include <iostream>  	

 

/*************************************************		  
 * Honor pledge (please write your name.) 
 *  	
 * I **Scott Forrest** have completed this code myself, without 
 * unauthorized assistance, and have followed the academic honor code. 
 * 
 * Edit the code below to complete your solution.
 *
 ***************************************************
 */


struct linkedlist  	
{
    int value;		  
	linkedlist* next;		  
};		  


int main(int argc, const char** argv)		  
{		  
	// Setup a root node pointer and a temporary variable n
	linkedlist* node = 0; 
	int n;		  

	// Push each value onto the *front* of the linked list
	while (std::cin >> n)
	{
			// Save ptr to the old root as variable next
			linkedlist* next = node;
			node = new linkedlist();
			node->next = next;
			node->value = n;
	}

	// Print each value in the linked list, in order
	for (linkedlist* cur = node; cur != 0; cur = cur->next)
			std::cout << cur->value << std::endl;

	// Cleanup each link when finished  	
	for (linkedlist* cur = node; cur != 0; cur = cur->next) 
		delete cur; 

	// Exit with error code 0 (success)	
	return 0;  	
}


